src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -214,7 +232,7 @@ protected virtual async Task ExecuteNodeAsync(ExecutionContext context, Executio
         {
             context.CancellationToken.ThrowIfCancellationRequested();
 
-            if (node.IsResultSet)
+            if (node is RootExecutionNode || node.Parent is ArrayExecutionNode)
```

Comment:
These are the only conditions upon which a node has already been executed when this method is called.  It is up to the execution strategy to not call this method twice for the same node.

Reply:
Please add these comments above that check.

Reply:
ping

Reply:
```suggestion
            // these are the only conditions upon which a node has already been executed when this method is called
            if (node is RootExecutionNode || node.Parent is ArrayExecutionNode)
```